pkg/ovs/mcp/mcp.go

```diff
@@ -0,0 +1,377 @@
+	in ovstypes.ShowParams) (*mcp.CallToolResult, ovstypes.ShowResult, error) {
+	result := ovstypes.ShowResult{}
+
+	// Run ovs-vsctl show command
+	lines, err := s.runCommand(ctx, req, in.NamespacedNameParams, []string{"ovs-vsctl", "show"})
+	if err != nil {
+		return nil, result, fmt.Errorf("failed to retrieve ovs configuration from pod %s/%s: %w",
+			in.Namespace, in.Name, err)
+	}
+
+	// Limit to MaxLines if specified
+	lines = limitLines(lines, in.MaxLines)
+
+	// Join all lines into a single output string
+	result.Output = strings.Join(lines, "\n")
+	return nil, result, nil
+}
+
+func (s *MCPServer) ListPorts(ctx context.Context, req *mcp.CallToolRequest,
+	in ovstypes.GetOVSCommandParams) (*mcp.CallToolResult, ovstypes.PortResult, error) {
+	result := ovstypes.PortResult{}
+
+	// Validate bridge name
+	if err := validateBridgeName(in.Bridge); err != nil {
+		return nil, result, err
+	}
+
+	// Run ovs-vsctl list-ports command
+	ports, err := s.runCommand(ctx, req, in.NamespacedNameParams, []string{"ovs-vsctl", "list-ports", in.Bridge})
+	if err != nil {
+		return nil, result, fmt.Errorf("failed to retrieve ports for bridge %s from pod %s/%s: %w",
+			in.Bridge, in.Namespace, in.Name, err)
+	}
+	result.Ports = append(result.Ports, ports...)
+	return nil, result, nil
+}
+
+func (s *MCPServer) ListInterfaces(ctx context.Context, req *mcp.CallToolRequest,
+	in ovstypes.GetOVSCommandParams) (*mcp.CallToolResult, ovstypes.InterfaceResult, error) {
+	result := ovstypes.InterfaceResult{}
+
+	// Validate bridge name
+	if err := validateBridgeName(in.Bridge); err != nil {
+		return nil, result, err
+	}
+
+	// Run ovs-vsctl list-ifaces command
+	ports, err := s.runCommand(ctx, req, in.NamespacedNameParams, []string{"ovs-vsctl", "list-ifaces", in.Bridge})
+	if err != nil {
+		return nil, result, fmt.Errorf("failed to retrieve interfaces for bridge %s from pod %s/%s: %w",
+			in.Bridge, in.Namespace, in.Name, err)
+	}
+	result.Interfaces = append(result.Interfaces, ports...)
+	return nil, result, nil
+}
+
+// DumpFlows dumps flows from a specific OVS bridge.
+func (s *MCPServer) DumpFlows(ctx context.Context, req *mcp.CallToolRequest,
+	in ovstypes.GetOVSCommandParams) (*mcp.CallToolResult, ovstypes.FlowsResult, error) {
+	result := ovstypes.FlowsResult{
+		Bridge: in.Bridge,
+	}
+
+	// Validate bridge name
+	if err := validateBridgeName(in.Bridge); err != nil {
+		return nil, result, err
+	}
+
+	// Run ovs-ofctl dump-flows command
+	flows, err := s.runCommand(ctx, req, in.NamespacedNameParams, []string{"ovs-ofctl", "dump-flows", in.Bridge})
+	if err != nil {
+		return nil, result, fmt.Errorf("failed to dump flows for bridge %s on pod %s/%s: %w",
+			in.Bridge, in.NamespacedNameParams.Namespace, in.NamespacedNameParams.Name, err)
+	}
+
+	// Filter flows by pattern if provided
+	flows, err = filterLines(flows, in.Filter)
+	if err != nil {
+		return nil, result, err
+	}
+
+	// Limit to MaxLines if specified
+	flows = limitLines(flows, in.MaxLines)
```

Comment:
Shold we validate `MaxLines` is not negative? Since providing zero allows the option to get all the lines from the command execution, does it make sense to not allow negative values at all?

Reply:
This applies to the other tools with `MaxLines` option.

Reply:
sure, now restricting output with only 100 lines when in.MaxLines <= 0, i think that make sense (as same as sos commands).